featureflags.go

```diff
@@ -94,91 +141,454 @@ func (poller *FeatureFlagsPoller) fetchNewFeatureFlags() {
+			return true, nil
+		} else {
+			return false, nil
+		}
+	}
+
+	if operator == "not_regex" {
+		var r *regexp.Regexp
+		var err error
+
+		if valueString, ok := value.(string); ok {
+			r, err = regexp.Compile(valueString)
+		} else if valueInt, ok := value.(int); ok {
+			valueString = strconv.Itoa(valueInt)
+			r, err = regexp.Compile(valueString)
+		} else {
+			errMessage := "Regex expression not allowed"
+			return false, errors.New(errMessage)
+		}
+
+		// invalid regex
+		if err != nil {
+			return false, nil
+		}
+
+		var match bool
+		if valueString, ok := override_value.(string); ok {
+			match = r.MatchString(valueString)
+		} else if valueInt, ok := override_value.(int); ok {
+			valueString = strconv.Itoa(valueInt)
+			match = r.MatchString(valueString)
+		} else {
+			errMessage := "Value type not supported"
+			return false, errors.New(errMessage)
+		}
+
+		if !match {
+			return true, nil
+		} else {
+			return false, nil
+		}
+	}
+
+	if operator == "gt" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
+		}
+
+		return overrideValueOrderable > valueOrderable, nil
+	}
+
+	if operator == "lt" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
 		}
+
+		return overrideValueOrderable < valueOrderable, nil
 	}
 
-	if featureFlag.Key == "" {
-		return defaultResult, nil
+	if operator == "gte" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
+		}
+
+		return overrideValueOrderable >= valueOrderable, nil
 	}
 
-	return poller.getFeatureFlagVariant(featureFlag, key, distinctId)
+	if operator == "lte" {
+		valueOrderable, overrideValueOrderable, err := validateOrderable(value, override_value)
+		if err != nil {
+			return false, err
+		}
+
+		return overrideValueOrderable <= valueOrderable, nil
+	}
+
+	return false, nil
+
+}
+
+func validateOrderable(firstValue interface{}, secondValue interface{}) (float64, float64, error) {
+	convertedFirstValue, err := interfaceToFloat(firstValue)
+
+	if err != nil {
+		errMessage := "Value 1 is not orderable"
+		return 0, 0, errors.New(errMessage)
+	}
+	convertedSecondValue, err := interfaceToFloat(secondValue)
+
+	if err != nil {
+		errMessage := "Value 2 is not orderable"
+		return 0, 0, errors.New(errMessage)
+	}
+
+	return convertedFirstValue, convertedSecondValue, nil
+
+}
+
+func interfaceToFloat(val interface{}) (float64, error) {
+
+	var i float64
+	switch t := val.(type) {
+	case int:
+		i = float64(t)
+	case int8:
+		i = float64(t)
+	case int16:
+		i = float64(t)
+	case int32:
+		i = float64(t)
+	case int64:
+		i = float64(t)
+	case float32:
+		i = float64(t)
+	case float64:
+		i = float64(t)
+	case uint8:
+		i = float64(t)
+	case uint16:
+		i = float64(t)
+	case uint32:
+		i = float64(t)
+	case uint64:
+		i = float64(t)
+	default:
```

Comment:
just checking, not sure about go syntax: This doesn't always go to `default` right? (it would in other langs, because there's no break/return in the cases above)

Reply:
nope!